afl-fuzz.c

```diff
@@ -5980,7 +5980,7 @@ static u8 fuzz_one(char** argv) {
   stage_name  = "user extras (insert)";
   stage_short = "ext_UI";
   stage_cur   = 0;
-  stage_max   = extras_cnt * len;
+  stage_max   = extras_cnt * (len + 1);
```

Comment:
would it be better to fix the loop condition instead (line 5989) ?

Reply:
> No. It would make it inconsistent with rest of the afl code.

As I check the similar loops both above and below this one, they both are iterating up to `i < len`, not `i <= len`, so most likely the error is there in the loop exit condition.

/cc @jonathanmetzman and @lszekeres for another pair of eyes

Reply:
I sent this patch to `afl` author two years ago. It is still broken and not fixes. A delay on this PR is 6 months already. And it is a trivial change. It is really discouraging to contribute any fixes to project managed like this.

The fix is correct. And consistent with rest of the code. Changing the loop condition on line 5989 would be a functional change that breaks the fuzzer (it is for insertions, and insertions can be from position 0 to len inclusive. That is before first byte, and after last byte.

The consistency comes from how you count number of iterations in loops based on initialization and end condition.

Reply:
My bad, sorry for the delay, I totally let his PR fell through the cracks! :/ 

Re @Dor1s's question, I think the reason this loop has `len+1` iterations is that it's an insertion phase and we insert the dictionary item at the very end of the input in the last iteration (at iteration `len+1`). So the patch LGTM. 

Thanks a lot @baryluk for the fix and for your patience!